src/ScribuntoLuaLibrary.php

```diff
@@ -56,7 +56,9 @@ public function getQueryResult( $queryString = null ) {
 		$api = new ApiMain( $params );
 		$api->execute();
 		$data = $api->getResult()->getResultData();
-
+		if(!empty($data) && !empty($data["query"]) && !empty($data["query"]["results"])) {
+			$data["query"]["results"] = array_combine(range(1, count($data["query"]["results"])), array_values($data["query"]["results"]));
```

Comment:
Once the first point has been taken care of I will merge this with reservation as it has no tests (and the behaviour changes here) but instead of blocking it,  #7 has been created as follow-up.